imap_processing/lo/l0/science_direct_events.py

```diff
@@ -0,0 +1,202 @@
+"""L1A Science Direct Events data class."""
+from dataclasses import dataclass
+from itertools import compress
+
+import bitstring
+from bitarray import bitarray
+
+import imap_processing.lo.l0.compression_tables as ct
+from imap_processing.ccsds.ccsds_data import CcsdsData
+from imap_processing.lo.l0.lol0 import LoL0
+
+
+@dataclass
+class ScienceDirectEvents(LoL0):
+    """L1A Science Drect Events data.
+
+    The Science Direct Events class handles the parsing and
+    decompression of L0 to L1A data.
+
+    Attributes
+    ----------
+    SHCOARSE : int
+        Spacecraft time.
+    COUNT: int
+        Number of direct events.
+    CHKSUM: int
+        Checksum for the packet.
+    DATA: bitstring.Bits
+        Compressed TOF Direct Event time tagged data.
+    TOF0: int
+        Time of Flight 0 value for direct event.
+    TOF1: int
+        Time of Flight 1 value for direct event.
+    TOF2: int
+        Time of Flight 2 value for direct event.
+    TOF3: int
+        Time of Flight 3 value for direct event.
+    TIME: int
+        time tag for the direct event
+    ENERGY: int
+        energy of the direct event ENA.
+    POS: int
+        Stop position for the direct event. There are 4 quadrants
+        on the at the stop position.
+    CKSM: int
+        This is checksum defined relative to the TOFs
+        condition for golden triples. If golden triples are below
+        a certain threshold in checksum it's considered golden, otherwise,
+        it's considered a silver triple. This is important for the compression
+        for golden triples because it's used to recover TOF1 because
+        compression scheme to save space on golden triples doesn't send
+        down TOF1 so it's recovered on the ground using the checksum
+    case_number: int
+        The compression case number for the direct event. The case number
+        determines how the bits are arranged in the compressed data.
+    tof_calculation_binary: dict
```

Comment:
Could you add a bit more about what the `dict` and `list`s here end up looking like? It's not clear to me after looking at where they are set.

Reply:
Also, you can be more specific in the typehint if that makes sense: `list[str]` or `dict[str, int]`

Reply:
Definitely. I'll add more info for those